addons/account/models/account_move.py

```diff
@@ -1505,7 +1505,11 @@ def _compute_tax_lock_date_message(self):
 
     @api.constrains('line_ids', 'journal_id')
     def _validate_move_modification(self):
-        if 'posted' in self.mapped('line_ids.payment_id.state'):
+        payment = self.mapped('line_ids.payment_id')
+        if payment.state == 'posted' and \
```

Comment:
it is not guarantied that `payment` is singleton, this line may raise an error, both self and lin_ids can containe multiple records

Reply:
> even within the same move? frowning_face

Uh? The constraint can be used to validate multiple moves at the same time.
But no, not within the same move since the payment is related on the move...

Reply:
Ok, as the current code is iterating over `self`, then `payment` should be a single record